File: SmartIndoorNavigation/src/main/java/com/Snoopy/SmartIndoorNavigation/Model/Entity/Product.java
package com.Snoopy.SmartIndoorNavigation.Model.Entity;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;



@Entity
public class Product {
	
	@Id
	@GeneratedValue
	private Long id;
	
	private String name;
	private double price;
	private String artNr;
	
	public Product() {
		
	}
	
	public Product(String name, double price, String artNr) {
		super();
		this.name = name;
		this.price = price;
		this.artNr = artNr;
	}
	
	public String getName() {
		return name;
	}
	public void setName(String name) {
		this.name = name;
	}
	public double getPrice() {
		return price;
	}
	public void setPrice(double price) {
		this.price = price;
	}
	public String getArtNr() {
		return artNr;
	}
	public void setArtNr(String artNr) {
		this.artNr = artNr;
	}
	
	
}

File: Verwaltungsfrontend/html/js/objects/Product.js
let productCounter = 0;

function Product(name, price, category) {
  this.id = productCounter++;
  this.name = name;
  this.price = price;
  this.category = category;

  this.getId = function() {
    return this.id;
  }

  this.setName = function(name) {
    this.name = name;
  }

  this.getName = function() {
    return this.name;
  }

  this.setPrice = function(price) {
    this.price = price;
  }

  this.getPrice = function() {
    return this.price;
  }

  this.setCategory = function(category) {
    this.category = category;
  }

  this.getCategory = function() {
    return this.category;
  }
}
